### PR TITLE
Add load/save preset

### PR DESCRIPTION
**Pull request by arbruijn**
New, simpler version without levels, just a file per preset. To bundle presets use a .dxa archive.

Depends on #26, otherwise seems ready for merging.
image
image
image

### CODE REVIEW

File: d1/main/net_udp.c [hunk 1]
```diff
@@ -4343,6 +4333,138 @@ int net_udp_setup_game()
...
+				return 0;
+			}
+			if (event_key_get(event) == KEY_CTRLED+KEY_D && citem >= 0) {
+				if (nm_messagebox(NULL, 2, TXT_YES, TXT_NO, "Delete preset %s?", items[citem]) != 0)
+					return 1;
+				snprintf(filename, sizeof(filename), "%s%s", items[citem], PRESET_EXT);
+				if (PHYSFS_delete(filename))
+					listbox_delete_item(lb, citem);
+				else
+					nm_messagebox(NULL, 1, TXT_OK, "Couldn't delete preset %s", TXT_COULDNT, TXT_DELETE_DEMO, items[citem]);
```
Comment (SiriusTR) [inline d1/main/net_udp.c:4362]: Might need a different string than TXT_DELETE_DEMO

**Review by SiriusTR — APPROVED** [review]
I think the string change for the failed profile write is the only thing that particularly needs addressing. Otherwise it looks good to go.

**Comment by SiriusTR** [comment]
Great - I see there were a few bug fixes too. Looks good!

**Comment by arbruijn** [comment]
Thank you, yes this should be it.

**Pull request merged**